src/components/PlaybackWrapper/playback.js

```diff
@@ -0,0 +1,147 @@
+import {
+  SERVICE_SPOTIFY,
+  SERVICE_YOUTUBE,
+  SERVICE_JUKEBOX_RADIO,
+} from '../../config/services';
+
+
+/*
+ * This gets the position that the track should be at.
+ *
+ * @return [
+ *   progress: Time in milliseconds,
+ *   seekTimeout: Time in milliseconds until the next seek should happen,
+ * ]
+ */
+export const getPositionMilliseconds = function(stream, startedAt) {
+  if(!stream.nowPlaying) {
+    return [undefined, undefined];
+  }
+
+  let progress = Date.now() - startedAt,
+      seekTimeout,
+      playbackIntervalIdx = 0,
+      cumulativeProgress = 0;
+
+  while(true) {
+    const playbackInterval = stream.nowPlaying.playbackIntervals[playbackIntervalIdx],
+          playbackIntervalDuration = playbackInterval[1] - playbackInterval[0],
+          remainingProgress = progress - cumulativeProgress;
+    if(remainingProgress < playbackIntervalDuration) {
+      progress = playbackInterval[0] + remainingProgress;
+      seekTimeout = playbackInterval[1] - progress;
+      break;
+    }
+    playbackIntervalIdx += 1;
+    cumulativeProgress += playbackIntervalDuration;
+  }
+
+  if(playbackIntervalIdx === stream.nowPlaying.playbackIntervals.length - 1) {
+    seekTimeout = undefined;
+  }
+
+  return [progress, seekTimeout];
+}
+
+
+/*
+ *
+ */
+export const playbackStart = function(playback, stream) {
```

Comment:
@op1346 these functions with all the `// TODO` comments are the main points here. I'm gonna merge in the PR now, but tagging you here to keep you in the loop.

Reply:
The `// TODO` will go in YouTube and MP3 playback logic.